### ISSUE TITLE
Section Background fails randomly

### ISSUE THREAD

**Issue by PzKfWg** [SharePoint/sp-dev-docs#8928]
Hi, I am using precisely the guidance from this "Supporting section backgrounds" page and find that the Theme Provider sends randomly the wrong (white) background color. 
https://github.com/SharePoint/sp-dev-docs/blob/main/docs/spfx/web-parts/guidance/supporting-section-backgrounds.md

If I just refresh my page, I get a white background 3 times out of 4. I even tried to do a 1M times loop if I get a #FFFFFF background color to try to get another value, but it does nothing. When I get a white background, the event handler does not fire either if I try to change the section background.
![image](https://user-images.githubusercontent.com/24944003/234994078-3f397478-66e8-42a9-b955-d603f8838eab.png)
![image](https://user-images.githubusercontent.com/24944003/234994116-163f38d1-b3af-4a83-a1a2-46dc90a70646.png)
(It works perfectly well on the Workbench, though)

In the debugger, I clearly see that the returned value from this._themeVariant.semanticColors.bodyBackground is #FFFFFF.

Is this a known bug and is there a work around?

Here is my code:
main ts.
```
import { ThemeProvider,ThemeChangedEventArgs,IReadonlyTheme } from '@microsoft/sp-component-base'; /*Background support*/
[...]
export default class BoralexHorlogesCompactesEtActionBoursiereWebPart extends BaseClientSideWebPart<IBoralexHorlogesCompactesEtActionBoursiereWebPartProps> {
  /*Background support
    https://learn.microsoft.com/en-us/sharepoint/dev/spfx/web-parts/guidance/supporting-section-backgrounds
  */
  private _themeProvider: ThemeProvider;
  private _themeVariant: IReadonlyTheme | undefined;

  protected onInit(): Promise<void> {    
    //My "clever" workaround to try to get another value if I get white
    let i = 0;    
    do {
      // Consume the new ThemeProvider service
      this._themeProvider = this.context.serviceScope.consume(ThemeProvider.serviceKey);
    
      // If it exists, get the theme variant
      this._themeVariant = this._themeProvider.tryGetTheme();
      i++;
      
    } while (this._themeVariant.semanticColors.bodyBackground === '#FFFFFF' && i<100000)
  
    // Register a handler to be notified if the theme variant changes
    this._themeProvider.themeChangedEvent.add(this, this._handleThemeChangedEvent);
  
    return super.onInit();
  }
  private _handleThemeChangedEvent(args: ThemeChangedEventArgs): void {
    this._themeVariant = args.theme;
    this.render();
  }
[...]
```

Component:

```
import { IReadonlyTheme } from '@microsoft/sp-component-base';
[...]

export default class BoralexHorlogesCompactesEtActionBoursiere extends React.Component<IBoralexHorlogesCompactesEtActionBoursiereProps, {}> {
  public render(): React.ReactElement<IBoralexHorlogesCompactesEtActionBoursiereProps> {    
    const { semanticColors }: IReadonlyTheme = this.props.themeVariant;
    return (
      <div id='BoralexHorlogesCompactesEtActionBoursiere' className={styles.boralexHorlogesCompactesEtActionBoursiere} style={{backgroundColor: semanticColors.bodyBackground}}>
        <Clocks TimeZones={this.props.TimeZones} displayMode={this.props.displayMode} />
        <StockTicker StockSymbol={this.props.StockSymbol} APIURL={this.props.APIURL} AllOrigin={this.props.AllOrigin} StockLink={this.props.StockLink} displayMode={this.props.displayMode} context={this.props.context} />
      </div>
    );
  }
[...]
```

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: d5fe832e-68cc-0f3a-acc1-c9116a5c981d
* Version Independent ID: 4b6961ad-45a6-aef9-f197-c31b1cb4a5f8
* Content: [Supporting section backgrounds](https://learn.microsoft.com/en-us/sharepoint/dev/spfx/web-parts/guidance/supporting-section-backgrounds)
* Content Source: [docs/spfx/web-parts/guidance/supporting-section-backgrounds.md](https://github.com/SharePoint/sp-dev-docs/blob/main/docs/spfx/web-parts/guidance/supporting-section-backgrounds.md)
* Service: **sharepoint**
* Sub-service: **sharepoint-framework**
* GitHub Login: @VesaJuvonen
* Microsoft Alias: **vesaj**

**Comment by AJIXuMuK** [comment]
@PzKfWg - thanks for reporting this one!
The documentation is a bit outdated.
Currently you don't need to specify event handler for the theme changed event.
You just need to override `protected onThemeChanged(currentTheme: IReadonlyTheme | undefined): void`.

Could you please try using that and let me know if that works?
For the example you can just scaffold a new web part solution using one of the latest SPFx versions.

**Comment by PzKfWg** [comment]
Thank you so much for your answer! This did the trick, I did not realize that the basic web part had already the code to correctly manage the background so I copied it from a new web part solution as suggested. This is way simpler compared to the old way!

Basically, the import in the TS is simply:
`import { IReadonlyTheme } from '@microsoft/sp-component-base';`

Then, no need for a Init method, only this method is needed:
```
  protected onThemeChanged(currentTheme: IReadonlyTheme | undefined): void {
    if (!currentTheme) {
      return;
    }
    
    const {
      semanticColors
    } = currentTheme;

    if (semanticColors) {
      this.domElement.style.setProperty('--bodyText', semanticColors.bodyText || null);
      this.domElement.style.setProperty('--link', semanticColors.link || null);
      this.domElement.style.setProperty('--linkHovered', semanticColors.linkHovered || null);
    }
  }
```
(Not sure what the last part does)
You don't need either to send the theme as a prop to the components: as long as you inherit from the parent background color, there is no need to specify dynamically the style which was cumbersome.

Thank you again! I don't know how we could update the documentation in a meaningful way because there is pretty much nothing to do except keep the default code from the generator...?
